src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1123,28 +1123,15 @@ internal override void CreateAsync()
                         _processInstance.RunspacePool.Dispose();
                     }
 
-                    stdInWriter = _processInstance.StdInWriter;
-                    // if (stdInWriter == null)
-                    {
-                        _serverProcess.OutputDataReceived += new DataReceivedEventHandler(OnOutputDataReceived);
-                        _serverProcess.ErrorDataReceived += new DataReceivedEventHandler(OnErrorDataReceived);
-                    }
-
                     _serverProcess.Exited += new EventHandler(OnExited);
-
-                    // serverProcess.Start();
                     _processInstance.Start();
 
-                    if (stdInWriter != null)
+                    stdInWriter = _processInstance.StdInWriter;
+                    if (stdInWriter == null)
```

Comment:
Actually, after discussion with @daxian-dbw , this null check should not be here, because we need to start our listening threads regardless of whether we create the host process or it is passed in.

Reply:
The check is removed.